lib/command/index.js

```diff
@@ -372,11 +372,21 @@ class Command extends Registry {
       const arg = toQuestion(flag, current, answers)
       const res = await this.prompt(arg)
       Object.assign(answers, res)
+      // If the flag has a validation function, call it now to possibly terminate early.
+      // Pass the shape of `this.parsed` to match the parameter from non-interactive mode.
+      if (typeof current.validate === 'function') {
+        const is_valid = current.validate.call(this, {...this.parsed, ...answers})
+        throwIfFlagFailedValidation(flag, is_valid)
+      }
     }
 
     this.parsed = {...this.parsed, ...answers}
 
-    for (const [flag, answer] of Object.entries(answers)) {
+    for (const [flag, user_answer] of Object.entries(answers)) {
+      const answer = user_answer === ''
+        ? undefined
+        : user_answer
```

Comment:
This is probably a notable breaking change

Reply:
Gotcha.  Yeah, that was necessary.

Reply:
Added to commit footer